egs2/musdb18/enh1/local/data.sh

```diff
@@ -0,0 +1,176 @@
+        num_train_audios=$(find "${MUSDB18}/train" -iname "*${format}" | wc -l)
+    fi
+    if [ "${num_test_audios}" = "${num_test}" ] && [ "${num_train_audios}" = "${num_train}" ]; then
+        echo "'${MUSDB18}/' already exists. Skipping..."
+    else
+        wget --continue -O "${MUSDB18}/musdb18.zip" ${musdb18_url}
+        unzip "${MUSDB18}/musdb18.zip" -d "${MUSDB18}"
+    fi
+
+    if ! ${use_hq}; then
+        log "Converting mp4 to wav format..."
+
+        if ! command -v ffmpeg &> /dev/null; then
+            log "Please install ffmpeg first"
+            exit 1
+        fi
+        # convert multi-stream mp4 to multiple wav audios
+        for dset in train test; do
+            find "${MUSDB18}/${dset}" -iname "*${format}" | while read -r fname; do
+                fbasename=$(basename "${fname}" .stem.mp4)
+                outdir="${cdir}/data/musdb18/${dset}/${fbasename}"
+                mkdir -p "${outdir}"
+                ffmpeg -nostdin -hide_banner -loglevel warning -i "${fname}" \
+                    -map 0:a:0 -c pcm_s16le "${outdir}/mixture.wav" \
+                    -map 0:a:1 -c pcm_s16le "${outdir}/drums.wav" \
+                    -map 0:a:2 -c pcm_s16le "${outdir}/bass.wav" \
+                    -map 0:a:3 -c pcm_s16le "${outdir}/other.wav" \
+                    -map 0:a:4 -c pcm_s16le "${outdir}/vocals.wav"
+            done
+        done
+    else
+        mkdir -p "${cdir}/data"
+        ln -s "$MUSDB18" "${cdir}/data/musdb18"
+    fi
+fi
+
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "stage 1: Data preparation"
+
+    for x in train test; do
+        dset="${x}_${sample_rate}"
+        mkdir -p "${cdir}/data/${dset}"
+        find "${cdir}/data/musdb18/${x}" -iname "mixture.wav" > "${cdir}/data/${dset}/flist"
+        awk -F"/" '{print(tolower($(NF-1)))}' "${cdir}/data/${dset}/flist" | sed -e 's/ /_/g' > "${cdir}/data/${dset}/uid"
+        if [ $nchannels -eq 1 ]; then
+            sed -i -e 's/^\(.*\)/sox "\1" -t wav - remix 1 |\nsox "\1" -t wav - remix 2 |/g' "${cdir}/data/${dset}/flist"
+            sed -i -e 's/^\(.*\)/\1_CH1\n\1_CH2/g' "${cdir}/data/${dset}/uid"
+        fi
+        paste -d' ' "${cdir}/data/${dset}/uid" "${cdir}/data/${dset}/flist" | sort -u > "${cdir}/data/${dset}/wav.scp"
+        awk '{print($1 " " $1)}' "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/utt2spk"
+        utils/utt2spk_to_spk2utt.pl "${cdir}/data/${dset}/utt2spk" > "${cdir}/data/${dset}/spk2utt"
+
+        sed -e "s#/mixture.wav#/drums.wav#g" "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/spk1.scp"
+        sed -e "s#/mixture.wav#/bass.wav#g" "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/spk2.scp"
+        sed -e "s#/mixture.wav#/vocals.wav#g" "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/spk3.scp"
+        sed -e "s#/mixture.wav#/other.wav#g" "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/spk4.scp"
```

Comment:
Is it possible to add a readme in this recipe, where we can discuss the mapping

Reply:
@Emrys365, please do so in your follow up PR.

Reply:
Sure.